image/qcow2/qcow2.go

```diff
@@ -903,8 +905,9 @@ func readZero(p []byte, off int64, sz uint64) (int, error) {
 		}
 		err = io.EOF
 	}
-	for i := 0; i < l; i++ {
-		p[i] = 0
+	var n int
+	for n < l {
+		n += copy(p[n:l], zeroBuffer)
```

Comment:
n:l may exceed the length of zeroBuffer

Reply:
copy() handle slices of different size. It will copy `min(len(dst), len(src))`.
https://pkg.go.dev/builtin#copy